src/transformers/models/bloom/modeling_bloom.py

```diff
@@ -706,14 +804,26 @@ def forward(
         else:
             attention_mask = attention_mask.to(hidden_states.device)
 
-        alibi = build_alibi_tensor(attention_mask, self.num_heads, dtype=hidden_states.dtype)
+        alibi = build_alibi_tensor(attention_mask, self.num_heads)
 
         causal_mask = self._prepare_attn_mask(
             attention_mask,
             input_shape=(batch_size, seq_length),
             past_key_values_length=past_key_values_length,
         )
 
+        if hasattr(self, "tp_rank"):
```

Comment:
That's a terrible way to test IMO, let's store process_group as an attribute and just run `self.process_group is not None`

Reply:
Also rename process_group to `tp_process_group`

Reply:
I don't like `tp_process_group` as a name. The object is a `process_group`, `tp` is what we use it for, but by itself the object is NOT something related to TP. so the  variable should be named in my opinion by what it is, not what your local intentions are to do with it  (in general).

`self.tokenizer` >  `self.string_to_ids_fn`
(Because for instance `self.tokenizer.decode` is much more readable than `self.string_to_ids_fn.decode`

This is style so preferences may vary. And it's not always valid, but I found it's the best first approximation.

Reply:
>  run self.process_group is not None

We are in the `__init__` function, the object is not created at this point (it actually is but let's ignore that route because __new__ is another can of worms).